presto-geospatial/src/main/java/com/facebook/presto/geospatial/BingTileFunctions.java

```diff
@@ -206,8 +206,8 @@ private static BingTile latitudeLongitudeToTile(double latitude, double longitud
         double y = 0.5 - Math.log((1 + sinLatitude) / (1 - sinLatitude)) / (4 * Math.PI);
 
         int mapSize = mapSize(zoomLevel);
-        int tileX = (int) clip(x * mapSize + 0.5, 0, mapSize - 1);
-        int tileY = (int) clip(y * mapSize + 0.5, 0, mapSize - 1);
+        int tileX = (int) clip(x * mapSize, 0, mapSize - 1);
```

Comment:
In `tileXYToLatitudeLongitude` above we still use `0.5` in our calculations, is that calculation correct after this change? It would be nice to add some round trip tests that first creates the tiles with `latitudeLongitudeToTile` and then you feed the output to `tileXYToLatitudeLongitude` to see whether you end up with the same latitude and longitude.

Reply:
@nezihyigitbasi , this 0.5 corresponds to the following computation in latitudeLongitudeToTile:

double y = 0.5 - Math.log((1 + sinLatitude) / (1 - sinLatitude)) / (4 * Math.PI);

I'm not quite sure how to write a round trip test because latitudeLongitudeToTile is a many to one mapping where all points within a tile map to that tile while tileXYToLatitudeLongitude returns 4 corners of the tile.

Reply:
@nezihyigitbasi , the range of latitude is [-90, 90] while the corresponding range of y is [1, 0]. 0.5 is used to convert [0, 1] range to [-0.5, 0.5] range which is aligned with [-90, 90] and need only proper scaling.